### PR TITLE
[Merged by Bors] - feat: port Order.Heyting.Basic

### PR DESCRIPTION
**Pull request by kim-em**
mathlib3 sha 4e42a9d0a79d151ee359c270e498b1a00cc6fa4e 

- [x] depends on: #792 
- [x] depends on: https://github.com/leanprover-community/mathlib4/pull/800

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#792
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#792~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#792~~
* ~~leanprover-community/mathlib4#800~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Order/Heyting/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,1363 @@
...
+#align has_le.le.disjoint_compl_left LE.le.disjoint_compl_left
+
+theorem LE.le.disjoint_compl_right (h : a ≤ b) : Disjoint a (bᶜ) :=
+  disjoint_compl_right.mono_left h
+#align has_le.le.disjoint_compl_right LE.le.disjoint_compl_right
+
+theorem IsCompl.compl_eq (h : IsCompl a b) : aᶜ = b :=
+  sorry
+  -- Hopefully will work in nightly-2022-11-30
+  -- h.1.le_compl_left.antisymm' <| Disjoint.le_of_codisjoint disjoint_compl_left h.2
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Heyting/Basic.lean:844]: Doesn't seem like it

Comment (ChrisHughes24) [reply Mathlib/Order/Heyting/Basic.lean:844]: This is a very mysterious 
```
failed to synthesize instance
  BoundedOrder α
```
I can't make it into an MWE

Comment (hrmacbeth) [reply Mathlib/Order/Heyting/Basic.lean:844]: Here is an MWE, I think:
```lean
variable {α : Type _}

class BoundedOrder (α : Type u) [LE α]

class DistribLattice (α) extends LE α 

def Disjoint [DistribLattice α] (a b : α) : Prop := sorry

def Codisjoint [DistribLattice α] (a b : α) : Prop := sorry

theorem Disjoint.le_of_codisjoint [DistribLattice α] [BoundedOrder α] {a b c : α}
    (hab : Disjoint a b) (hbc : Codisjoint b c) : a ≤ c :=
  sorry

class HeytingAlgebra (α : Type _) extends LE α

instance (priority := 100) HeytingAlgebra.toDistribLattice [HeytingAlgebra α] : DistribLattice α :=
  sorry

instance (priority := 100) HeytingAlgebra.toBoundedOrder [HeytingAlgebra α] : BoundedOrder α :=
  sorry

example [HeytingAlgebra α] {a b c : α} (h₁ : Disjoint a b) (h₂ : Codisjoint b c) : a ≤ c :=
  Disjoint.le_of_codisjoint h₁ h₂
```

Comment (ChrisHughes24) [reply Mathlib/Order/Heyting/Basic.lean:844]: Changing the `sorry` for both of your instances into `{  }` seems to fix the problem in your MWE.

Comment (hrmacbeth) [reply Mathlib/Order/Heyting/Basic.lean:844]: Then probably I simplified too much, sorry!

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**